src/main/cpp/subsystems/Arm.cpp

```diff
@@ -45,61 +127,66 @@ bool Arm::ManualControl() {
     OKC_CHECK(interface_ != nullptr);
 
     interface_->arm_lift_power = lift_power_;
-    interface_->arm_up_power = up_power_;
     interface_->arm_extend_power = extend_power_;
 
-    arm_lift_output_log_.Append(interface_->arm_lift_power);
-    arm_lift_enc_log_.Append(interface_->arm_lift_encoder_val);
-
-
     return true;
 }
-bool Arm::SetManualUpPower(double power)
-{
+
+bool Arm::SetManualUpPower(double power) {
     up_power_ = power;
 
     return true;
 
 }
 
+bool Arm::AutoControl() {
+    OKC_CHECK(interface_ != nullptr);
+    OKC_CHECK(this->arm_pid_ != nullptr);
+
+    // if the setpoint is 0 (ie the arm hasn't been set to go somewhere yet, like it's just been enabled)
+    if (this->arm_pid_->GetSetpoint() == 0) {
+        // don't do anything
+        this->interface_->arm_lift_power = 0;
+    } else { // otherwise
+        // PID to the setpoint
+        this->interface_->arm_lift_power = this->arm_pid_->Calculate(this->interface_->arm_duty_cycle_encoder);
+    }
+
+    // this doesn't apply to extend because it is unlikely we would break something with this *knock on wood*
+    this->interface_->arm_extend_power = this->inches_pid_->Calculate(this->interface_->arm_extend_encoder);
```

Comment:
I hope `arm_extend_encoder` is a double/some numeric type and not an actual encoder

Reply:
correct.